### ISSUE TITLE
Insiders: Must authenticate a second time for Settings Sync

### ISSUE THREAD

**Issue by StevieDicks** [microsoft/vscode#148168]
Follow up to https://github.com/microsoft/vscode/issues/148055#issue-1214704251 ... Requested by @TylerLeonhardt to make a new issue RE: https://github.com/microsoft/vscode/issues/148055#issuecomment-1109903448

Note: This only happens upon first sign-in. On subsequent sign-ins it is not reproducible. It seems my credentials are stored in Settings Sync after the first sign-in, and it automatically turns on. 

Issue description: Although I am logged into GitHub on VS Code, Settings Sync forces authentication a second time. If I do the method of authentication linked in step 1 (below) then it works, but using the default does not work.

Steps to Reproduce:

1. Use the log-in method mentioned here: https://github.com/microsoft/vscode/issues/147104#issuecomment-1109108734 
    - Start here:
    - ![image](https://user-images.githubusercontent.com/50631116/165333542-fc5c7eb9-a62e-4162-a801-223930ee21d9.png)
    - Sign in with GitHub: 
    - ![image](https://user-images.githubusercontent.com/50631116/165333754-37b44855-36ee-4b7f-8a92-ba0a7ae7c121.png)
    - Click Cancel here: 
    - ![image](https://user-images.githubusercontent.com/50631116/165333834-0da4e6c1-c751-486f-a7af-b4b638e42154.png)
    - Choose to "try a different way":
    - ![image](https://user-images.githubusercontent.com/50631116/165334158-f0d8a1f0-44b8-44f2-9405-9c6be71ec9e9.png)
    - Click "Copy and Continue to GitHub"
    - ![image](https://user-images.githubusercontent.com/50631116/165334363-18077f07-278f-42e1-a69f-7fea177bf393.png)
    - Paste the code in the prompt on GitHub and continue. It should log you in

2. Now we are logged in but Settings Sync is not turned on. **(Only happens on first log in.)**
    - See here: 
    - ![image](https://user-images.githubusercontent.com/50631116/165334849-6ff46f45-5c16-4541-bde1-52abc58df2a9.png)
    - Click "Turn on Settings Sync..."
    - At Command Prompt, choose your options and click "Sign in and Turn on"
    - ![image](https://user-images.githubusercontent.com/50631116/165335068-6f06a7b2-d599-4ca7-b4ef-9602b59453dd.png)
    - Choose Insiders when prompted
    - If this is the first time signing in, you will be prompted to authenticate once again. The default method will not work, but the method linked in Step 1 will.

@TylerLeonhardt 

Does this issue occur when all extensions are disabled?: Yes

VS Code version: Code - Insiders 1.67.0-insider (ab86e0229d6b4d0cb49cfd6747c92cafcd2bd4af, 2022-04-26T05:16:02.333Z)
OS version: Linux x64 4.14.275-207.503.amzn2.x86_64
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Xeon(R) Platinum 8259CL CPU @ 2.50GHz (2 x 2499)|
|GPU Status|2d_canvas: unavailable_software<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: disabled_software<br>multiple_raster_threads: disabled_off<br>oop_rasterization: disabled_off<br>opengl: disabled_off<br>rasterization: disabled_software<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: unavailable_software<br>webgl2: unavailable_software|
|Load (avg)|0, 0, 0|
|Memory (System)|3.79GB (0.50GB free)|
|Process Argv|.|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|mate|
|XDG_CURRENT_DESKTOP|MATE|
|XDG_SESSION_DESKTOP|undefined|
|XDG_SESSION_TYPE|undefined|
</details>Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30422396
pythontb:30258533
pythonvspyt551:30291412
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
vscod805cf:30301675
pythonvspyt200:30323110
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30404738
wslgetstarted:30449409
pythonvspyt640:30438691
vscscmwlcmt:30465136
cppdebugcf:30475217

```

</details>

**Comment by sandy081** [comment]
@TylerLeonhardt Is this auth related issue or shall settings sync has to do something here?

**Comment by TylerLeonhardt** [comment]
@sandy081 I don't think you have anything to do... unless there's some timeout for settings sync?

@StevieDicks there are suppose to be 2 flows that happen before the device code flow:

* The main flow that tries to redirect back to VS Code (fastest)
* The web flow that shows this page:
![image](https://user-images.githubusercontent.com/2644648/165638398-0260cc5d-f038-4c08-b86b-53f423c9dd29.png)
* Then the device code flow

I know the first one doesn't work on your system, but what about the 2nd? Do you see it at all? Btw, I did make some changes 2 days ago in this space so you may need to get the latest Insiders.

**Comment by StevieDicks** [comment]
@TylerLeonhardt I don't recall seeing the second option at all, only the first and third.

Also, I downloaded Insiders the day you mentioned it, so it should have been up to date with the changes you mentioned. Sorry I can't be of more help.

**Comment by TylerLeonhardt** [comment]
@StevieDicks I believe I made some updates after that... can you try again and let me know how it goes?

**Comment by StevieDicks** [comment]
Just updated Insiders and was able to see the second method as below. This works better than the device code flow. 

![image](https://user-images.githubusercontent.com/50631116/165829916-87b03e4b-a851-4167-9102-b8a7a9aca27f.png)

**Comment by StevieDicks** [comment]
I am still not sure if this addresses the settings sync, as I mentioned it only happens on the first login and I don't have the time to go through all the steps to reproduce that. But it is definitely a better sign in process!

**Comment by TylerLeonhardt** [comment]
I think I'm going to close this. I wasn't able to repro the initially reported problem and it only happens if you have to resort to device code flow which only happens after the 2 other (better) flows fail to happen. Given the scope is very tiny, I think there are bigger fish to fry.

@StevieDicks let me know if you really want me to keep this open, but given the severity and how minor the inconvenience is IMO, I don't think we really need to track this.

**Comment by StevieDicks** [comment]
@TylerLeonhardt, Thanks for looking into it. It doesn't bother me at all -- I only wanted to bring it to your attention for your record. After the first login it does not happen anymore so it's difficult even for me to reproduce. GitHub integration is running smoothly on Insiders now. Cheers.
